### PR TITLE
feat(sso): UI

### PR DESCRIPTION
**Pull request by olarclara**
I might need to change the API a bit when hooking things with the API but wanted to get most of the UI out of the way.

You can directly access this by opening `/signin?sso_mode=true`

|IDLE|
|:-:|
|xkrisk-3000 csb app_signin_sso_mode=true (2)|

|VALIDATING|
|:-:|
|xkrisk-3000 csb app_signin_sso_mode=true (1)|

|INVALID|
|:-:|
|xkrisk-3000 csb app_signin_sso_mode=true|

### CODE REVIEW

File: packages/app/src/app/pages/SignIn/ChooseProvider.tsx [hunk 1]
```diff
@@ -8,23 +17,198 @@ import {
...
+      as={motion.div}
+      align="center"
+      initial={{ opacity: 0 }}
+      animate={{ opacity: 1 }}
+      exit={{ opacity: 0 }}
+      direction="vertical"
+      gap={8}
+    >
+      <StyledHeading as="h1" id="heading">
+        Sign in with SSO login
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/SignIn/ChooseProvider.tsx:82]: I've been looking at some other SSO examples and the text looks a bit duplicated to me if you pronounce SSO in full form: Sign in with Single Sign On Login. Should we change it to something like "Login with SSO"?

File: packages/app/src/app/pages/SignIn/ChooseProvider.tsx [hunk 2]
```diff
@@ -8,23 +17,198 @@ import {
...
+  const [validationState] = React.useState<ValidationState>({
+    state: 'IDLE',
+  });
+
+  React.useEffect(() => {
+    const inputElement = inputRef.current;
+    if (inputElement) {
+      inputElement.focus();
+    }
+  }, []);
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/SignIn/ChooseProvider.tsx:62]: Better to do this with the `autoFocus` attribute on the `input`.

File: packages/app/src/app/pages/SignIn/ChooseProvider.tsx [hunk 3]
```diff
@@ -8,23 +17,198 @@ import {
...
+        as="form"
+        css={{ width: '100%', maxWidth: '420px' }}
+        direction="vertical"
+        onSubmit={handleFormSubmit}
+        gap={2}
+      >
+        <Stack direction="vertical" gap={1}>
+          <Element css={{ position: 'relative', height: '48px' }}>
+            <Input
+              aria-describedby="heading error"
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/SignIn/ChooseProvider.tsx:94]:
```suggestion
              aria-labelledby="heading"
              aria-describedby="error"
```

There are some more things we can add or improve regarding a11y, but I would love to bake that into the components one day.

File: packages/app/src/app/pages/SignIn/ChooseProvider.tsx [hunk 4]
```diff
@@ -8,23 +17,198 @@ import {
...
 }) => {
-  const { loadingAuth, cancelOnLogin } = useAppState();
   const {
     signInButtonClicked,
     setLoadingAuth,
     toggleSignInModal,
   } = useActions();
+  const { loadingAuth, cancelOnLogin } = useAppState();
+  // const location = useLocation();
+  // const history = useHistory();
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/SignIn/ChooseProvider.tsx:207]: Can be removed?

File: packages/app/src/app/pages/SignIn/ChooseProvider.tsx (hunk 1 above)
Comment (lenasmayor) [reply packages/app/src/app/pages/SignIn/ChooseProvider.tsx:82]: Agree with Tristan

**Review by lenasmayor — APPROVED** [review]

**Pull request merged**